Modules/Common/include/Common/ReferenceComparatorTask.h

```diff
@@ -0,0 +1,123 @@
+///
+/// For each input plot, the task publishes the ratio between the plot and the corresponding reference.
+/// Moreover, for 1-D histograms it also publishes the plot itself with the reference superimposed, for visual comparison.
+///
+/// \author Andrea Ferrero
+class ReferenceComparatorTask : public quality_control::postprocessing::PostProcessingInterface
+{
+ public:
+  ReferenceComparatorTask() = default;
+  ~ReferenceComparatorTask() override = default;
+
+  void configure(const boost::property_tree::ptree& config) override;
+  void initialize(quality_control::postprocessing::Trigger, framework::ServiceRegistryRef) override;
+  void update(quality_control::postprocessing::Trigger, framework::ServiceRegistryRef) override;
+  void finalize(quality_control::postprocessing::Trigger, framework::ServiceRegistryRef) override;
+
+  struct HistoWithRef
+  {
+    std::shared_ptr<TH1> mPlot;
+    std::shared_ptr<TH1> mRefPlot;
+    std::shared_ptr<TCanvas> mCanvas;
+  };
+
+ private:
+  std::shared_ptr<o2::quality_control::core::MonitorObject> getRefPlot(o2::quality_control::repository::DatabaseInterface& qcdb, std::string fullPath, o2::quality_control::core::Activity activity);
+  void addHistoWithReference(std::shared_ptr<o2::quality_control::core::MonitorObject> refPlot, ReferenceComparatorTaskConfig::DataGroup group, std::string path);
+  void updatePlot(std::string plot, TObject* object);
+  void updatePlotRatio(std::string plot, TObject* object);
+
+  template <class HIST, class HISTRATIO>
+  void initHistograms1D(TObject* refObject,
+      const ReferenceComparatorTaskConfig::DataGroup& group,
+      std::string path);
+  template <class HIST, class HISTRATIO>
+  void initHistograms2D(TObject* refObject,
+      const ReferenceComparatorTaskConfig::DataGroup& group,
+      std::string path);
+
+  template <class T>
+  T getParameter(std::string parName, const T defaultValue, const o2::quality_control::core::Activity& activity);
+  template <class T>
+  T getParameter(std::string parName, const T defaultValue);
+
+  int mRefRun{ 0 };
+  int mNotOlderThan{ 120 };
+
+  /// \brief configuration parameters
+  ReferenceComparatorTaskConfig mConfig;
+  /// \brief list of plot names, separately for each group
+  std::map<std::string, std::vector<std::string>> mPlotNames;
+  /// \brief reference MOs
+  std::map<std::string, std::shared_ptr<o2::quality_control::core::MonitorObject>> mRefPlots;
+  /// \brief wether to auto-scale or not the reference of a given plot
+  std::map<std::string, bool> mRefScale;
+  /// \brief histograms with references superimposed
+  std::map<std::string, HistoWithRef> mHistograms;
+  /// \brief histograms with current/reference ratios
+  std::map<std::string, std::unique_ptr<TH1>> mHistogramRatios;
+};
+
+template <class T>
+T ReferenceComparatorTask::getParameter(std::string parName, const T defaultValue, const o2::quality_control::core::Activity& activity)
+{
+  T result = defaultValue;
```

Comment:
@Barthelemy would you consider adding this kind of template method directly into the `CustomParameters` class? I think it could be useful and cover a lot of use cases.